modules/nextflow/src/main/groovy/nextflow/processor/Arch.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023, Pawsey Supercomputing Research Centre
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.processor
+
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+
+/**
+ * Implements the {@code arch} process directive, to hold information on the 
+ * CPU (micro)architecture required by the process.
+ * 
+ * @author Marco De La Pierre <marco.delapierre@gmail.com>
+ */
+@ToString
+@EqualsAndHashCode
+@CompileStatic
+class Arch {
```

Comment:
Let's name this class `Architecture`, even if the directive is `arch`

Reply:
done!